src/main/java/org/fenixedu/academic/domain/BibliographicReference.java

```diff
@@ -52,6 +55,48 @@ public static BibliographicReference create(final String title, final String aut
         return result;
     }
 
+    public static BibliographicReference create(final LocalizedString title, final String authors,
+            final LocalizedString reference, final String year, final String url, final Integer referenceOrder,
+            final Boolean optional) {
+        if (title == null || authors == null || year == null || optional == null) {
+            throw new IllegalArgumentException("Required fields not filled");
+        }
+
+        final BibliographicReference result = new BibliographicReference();
+        result.setLocalizedTitle(title);
+        result.setAuthors(authors);
+        result.setLocalizedReference(reference);
+        result.setYear(year);
+        result.setUrl(url);
+        result.setOptional(optional);
+        result.setReferenceOrder(referenceOrder);
+
+        return result;
+    }
+
+    /**
+     * Copies the provided Bibliographic Reference.
+     * Note that this methods only copies the fields of this object. Not the relationships.
+     * 
+     * @param bibliographicReferenceToCopy
+     * @return copied BibliographicReference
+     */
+    public static BibliographicReference copy(BibliographicReference bibliographicReferenceToCopy) {
+
+        final Function<LocalizedString, LocalizedString> copyLocalizedString = ls -> {
+            LocalizedString copy = new LocalizedString(Locale.getDefault(), ls.getContent(Locale.getDefault()));
+            ls.getLocales().stream().filter(l -> l != Locale.getDefault()).forEach(l -> copy.with(l, ls.getContent(l)));
```

Comment:
Attention, the LocalizedString is a valueType and to ensure its immutability the method `with(..)` returns a new copy of the LocalizedString